### ISSUE TITLE
DL3006 rule fails in multi-stage build FROM

### ISSUE THREAD

**Issue by mbelang** [hadolint/hadolint#161]
- [x] This is a bug report

### Expected behavior
The rule should not fail when `FROM` base image uses an other stage in a multi-stage build Dockerfile

### Actual behavior
The rule fails

### Steps to reproduce the behavior

The second build stage `unit-test` of the following Dockerfile is using the previous stage as it's base image which is totally fine but the rule require that we specify a tag and no tag exists for a stage.

The Dockerfile is an example and the problem with the rule is not related to the language at all.

Using `hadolint/hadolint:latest` image to lint a multi-stage Dockerfile

```Dockerfile
FROM golang:1.9.3-alpine3.7 AS build

... (building my go app here)

FROM build as unit-test

... (testing my go app here)

FROM alpine:3.7
...
```

**Comment by mbelang** [comment]
Thanks for the fix.
In which version will it be released? I'm using your official docker image.

**Comment by lorenzo** [comment]
@mbelang it was release in v1.4.0
